feeds/wlan-ap/opensync/files/bin/wlan_ap_redirector.sh

```diff
@@ -1,11 +1,35 @@
 #!/bin/sh
 
-if [ $# -ne 1 ] ; then
-	echo "Usage: $0 <redirector address>" >&2
-	exit 1
+AP_PRIVATE_KEY_FILE="/usr/opensync/certs/client_dec.key"
+AP_CERTIFICATE_FILE="/usr/opensync/certs/client.pem"
+AP_DEVICE_ID_FILE="/usr/opensync/certs/client_deviceid.txt"
+DIGICERT_API_URI="clientauth.one.digicert.com"
+
+if [ "$1" = "-h" ]; then
+  echo "Usage: $0 [redirector address]" >&2
+  exit 1
 fi
 
-redirector_addr=$1
+# Query DigiCert's API if redirector wasn't specified
+if [ -z "$1" ]; then
+  if [ ! -f "$AP_DEVICE_ID_FILE" ]; then
+      echo "Device ID file $AP_DEVICE_ID_FILE does not exist. Make sure to create it or specify the redirector address manually."
+      exit 1
+  fi
+
+  # TODO: this command should be retried if it fails
+  digicert_device_id=`cat ${AP_DEVICE_ID_FILE}`
+  device_data=`curl -s \
+    --key "${AP_PRIVATE_KEY_FILE}" \
+    --cert "${AP_CERTIFICATE_FILE}" \
+    "https://${DIGICERT_API_URI}/iot/api/v2/device/${digicert_device_id}"`
+
+  controller_ip=`echo ${device_data} | jq '.fields | .[] | select ( .name == "Redirector" ) | .value' | tr -d '"'`
```

Comment:
@RickSommerville Do you know if `jq` is available on a stock firmware image?

Reply:
I don't think so.  I tried to just execute 'jq' from SSH to AP and it reports "-ash: jq: not found".
"which jq" similarly reports nothing.

Reply:
Mmh do you think it's feasible to include it in the image? Otherwise we will have trouble to reliably extract the redirector URL from the Digicert response.

Reply:
Or maybe you know of any other tool that is already installed and capable of handling JSON.

Reply:
Look at wan-ap-config/etc/uci-defaults/90-tip-hostname.  It uses "jsonfilter".  Relevant code is:

MAC=$(cat /etc/board.json | jsonfilter -e '@["network"]["wan"]["macaddr"]' | tr -d :)

Reply:
Awesome, thanks! Replaced the query using jsonfilter.